pkg/controller/client.go

```diff
@@ -116,47 +118,41 @@ func (c *XdsClient) clientResponseProcess(ctx context.Context) {
 			return
 		default:
 			if reconnect {
-				log.Warnf("reconnect due to %s", err)
-				if err = c.recoverConnection(); err != nil {
-					log.Errorf("recover connection failed, %s", err)
-					return
-				}
+				c.recoverConnection()
 				reconnect = false
 			}
 
 			if bpfConfig.AdsEnabled() {
-				if err = c.AdsStream.AdsStreamProcess(); err != nil {
+				if err = c.AdsStream.HandleAdsStream(); err != nil {
 					_ = c.AdsStream.Stream.CloseSend()
-					_ = c.grpcConn.Close()
-					reconnect = true
-					continue
 				}
 			} else if bpfConfig.WdsEnabled() {
-				if err = c.workloadStream.WorkloadStreamProcess(c.rbac); err != nil {
+				if err = c.workloadStream.HandleWorkloadStream(c.rbac); err != nil {
```

Comment:
Right now kmesh's mode defaults to workload mode. I think it's better to judge WdsEnable first.